fetcher/ubuntu.go

```diff
@@ -20,46 +21,63 @@ const (
 )
 
 // FetchUbuntuVulnList clones vuln-list and returns CVE JSONs
-func FetchUbuntuVulnList() (entries []models.UbuntuCVEJSON, err error) {
+func FetchUbuntuVulnList() (iter.Seq2[models.UbuntuCVEJSON, error], int, error) {
 	// Clone vuln-list repository
 	dir := filepath.Join(util.CacheDir(), "vuln-list")
 	updatedFiles, err := git.CloneOrPull(ubuntuRepoURL, dir, ubuntuDir)
 	if err != nil {
-		return nil, xerrors.Errorf("error in vulnsrc clone or pull: %w", err)
+		return nil, 0, xerrors.Errorf("error in vulnsrc clone or pull: %w", err)
 	}
 
 	// Only last_updated.json
 	if len(updatedFiles) <= 1 {
-		return nil, nil
+		return nil, 0, nil
 	}
 
 	rootDir := filepath.Join(dir, ubuntuDir)
 	targets, err := util.FilterTargets(ubuntuDir, updatedFiles)
 	if err != nil {
-		return nil, xerrors.Errorf("failed to filter target files: %w", err)
+		return nil, 0, xerrors.Errorf("failed to filter target files: %w", err)
 	} else if len(targets) == 0 {
 		log15.Debug("Ubuntu: no update file")
-		return nil, nil
+		return nil, 0, nil
 	}
 	log15.Debug(fmt.Sprintf("Ubuntu updated files: %d", len(targets)))
 
-	err = util.FileWalk(rootDir, targets, func(r io.Reader, _ string) error {
-		content, err := io.ReadAll(r)
-		if err != nil {
-			return err
-		}
+	count, err := countUbuntuCVEs(rootDir, targets)
+	if err != nil {
+		return nil, 0, xerrors.Errorf("failed to count Ubuntu CVEs: %w", err)
+	}
+
+	return func(yield func(models.UbuntuCVEJSON, error) bool) {
+
+		err = util.FileWalk(rootDir, targets, func(r io.Reader, _ string) error {
+			content, err := io.ReadAll(r)
+			if err != nil {
+				return err
+			}
 
-		cve := models.UbuntuCVEJSON{}
-		if err = json.Unmarshal(content, &cve); err != nil {
-			return xerrors.Errorf("failed to decode Ubuntu JSON: %w", err)
+			cve := models.UbuntuCVEJSON{}
+			if err = json.Unmarshal(content, &cve); err != nil {
+				return xerrors.Errorf("failed to decode Ubuntu JSON: %w", err)
+			}
+
+			if !yield(cve, nil) {
+				return err
```

Comment:
What is the expected err here?

Reply:
Thanks for catching it. It accidentally returns nil, causing [this bug](https://github.com/vulsio/gost/pull/291#pullrequestreview-2689321312).
It should have been fixed in https://github.com/vulsio/gost/pull/291/commits/f0b4d1c782d38b6e42455ff31851f9301a85613d.